internal/powerplatform/services/licensing/api_licensing.go

```diff
@@ -170,4 +190,47 @@ func (client *LicensingClient) RemoveEnvironmentsToBillingPolicy(ctx context.Con
 	_, err := client.Api.Execute(ctx, "POST", apiUrl.String(), nil, environments, []int{http.StatusOK}, nil)
 
 	return err
-}
\ No newline at end of file
+}
+
+func (client *LicensingClient) DoWaitForLifecycleOperationStatus(ctx context.Context, billingPolicyDto *BillingPolicyDto) (*BillingPolicyDto, error) {
```

Comment:
the method shouldn't be names ..."LifecycleOperationStatus" because it does not operate with that lifecycle service. You can just name it "DoWaitForFinalStatus" or something like that

Reply:
Method renamed with the proposition 😊